test/noauth.mk

```diff
@@ -0,0 +1,59 @@
+# Targets for testing the installer in 'no tls' mode.
+# This can be used for users who have ipsec or other secure VPC, or don't need the security features.
+# It is intended to verify that Istio can work without citadel for a-la-carte modes.
+
+# Included from the top level makefile, to keep things a bit more modular.
+
+# The tests run in a separate Kind cluster, ${KIND_CLUSTER}-noauth - otherwise citadel would create the certs.
+# Only a subset of features are supported - currently autoinject and validation are off until installer is able
+# to provision the certs they need to register with k8s.
+
+# Also used to test 'controlPlaneSecurity=OFF', for cases where citadel is not installed.
+
+# Will create a KIND cluster and run the tests in the docker image running KIND
+# You can also run the targets directly, against a real cluster: make run-test-noauth-full
+
+test-noauth:
+	$(MAKE) KIND_CLUSTER=${KIND_CLUSTER}-noauth maybe-clean maybe-prepare sync
+	$(MAKE) KIND_CLUSTER=${KIND_CLUSTER}-noauth kind-run TARGET="run-test-noauth-micro"
+	$(MAKE) KIND_CLUSTER=${KIND_CLUSTER}-noauth kind-run TARGET="run-test-noauth-full"
+
+# Run a test without authentication, minimal possible install. The cluster should have no certs (so we can
+# confirm that all the 'off' is respected.
+# Will install 2 control planes, one with only discovery + ingress ( micro ), one with galley, discovery, telemetry
+run-test-noauth-micro: install-crds
+	bin/iop ${ISTIO_NS}-micro istio-discovery ${BASE}/istio-control/istio-discovery ${IOP_OPTS} \
+		--set global.controlPlaneSecurityEnabled=false --set useMCP=false --set plugins="health"
+	kubectl wait deployments istio-pilot -n ${ISTIO_NS}-micro --for=condition=available --timeout=${WAIT_TIMEOUT}
+	bin/iop ${ISTIO_NS}-micro istio-ingress ${BASE}/gateways/istio-ingress --set global.istioNamespace=${ISTIO_NS}-micro \
+	 	${IOP_OPTS} --set global.controlPlaneSecurityEnabled=false
+	kubectl wait deployments ingressgateway -n ${ISTIO_NS}-micro --for=condition=available --timeout=${WAIT_TIMEOUT}
+
+# TODO: pass meshConfigFile, injectConfigFile, valuesFile to test, or skip the kube-inject and do it manually (better)
```

Comment:
Why can't it read the configmap, does it expect istio-system?

Reply:
There is no configmap and no inject template - since we don't install the injector. 
We don't install the injector because there are no certs ( and we want to verify a-la-carte - injector is in no way required ).